src/bootman/update.c

```diff
@@ -42,6 +42,47 @@ static int kernel_compare_reverse(const void *a, const void *b)
         return 1;
 }
 
+/**
+ * List kernels available on the target
+ *
+ * Returns a list of kernels available to be run.
+ */
+char **boot_manager_list_kernels(BootManager *self)
+{
+        assert(self != NULL);
+        autofree(KernelArray) *kernels = NULL;
+        autofree(char) *default_kernel = NULL;
+        char **results;
+
+        /* Grab the available kernels */
+        kernels = boot_manager_get_kernels(self);
+        if (!kernels || kernels->len == 0) {
+                LOG_ERROR("No kernels discovered in %s, bailing", self->kernel_dir);
+                return NULL;
+        }
+
+        /* Sort them to ensure static ordering */
+        nc_array_qsort(kernels, kernel_compare_reverse);
+
+        default_kernel = boot_manager_get_default_kernel(self);
+        results = calloc(kernels->len + (size_t)1, sizeof(char *));
+        if (!results) {
+                DECLARE_OOM();
+                return NULL;
+        }
+        for (uint16_t i = 0; i < kernels->len; i++) {
+                const Kernel *k = nc_array_get(kernels, i);
+                if (streq(default_kernel, k->meta.bpath)) {
```

Comment:
Is `streq` function from libnica? How does it take `NULL`?

Reply:
I think it should be fine, it basically wraps strcmp but has a check for `arg1 == arg2` and a `!arg1 || !arg2` before it and the NULL check returns false preventing NULL being passed to strcmp (but as the equal check runs first NULL for both args returns true).